src/lasdi/param.py

```diff
@@ -66,6 +79,33 @@ def createInitialTrainSpace(self, param_list):
         mesh_grids = self.createHyperMeshGrid(paramRanges)
         return self.createHyperGridSpace(mesh_grids)
     
+    def createInitialTrainSpaceForHull(self, param_list):
+        '''
+        If test_space is 'hull', then the provided training parameters must be 
+        points on the exterior of our training space. So, we form the provided points
+        into an array.
+        '''
+
+        paramRanges = []
+
+        k = 0
+        for param in param_list:
+            assert (param['test_space_type'] == 'exterior'), ('test_space_type for all parameters must '
+                                                            'be \'exterior\' when test_space is \'hull\'. ')
+            
+            _, paramRange = getParam1DSpace[param['test_space_type']](param)
```

Comment:
Essentially, this is `getParam1DSpace['list'](param)`. I don't think we need a separate function for this, especially when you don't use `Nx` in this loop. I suggest replacing line 96 into:
```
_, paramRange = getParam1DSpace['list'](param)
```

I see in the PR below you use `Nx`, which is essentially `getParam1DSpace['uniform'](param)`. It is fine to use both for `hull` test space, but to me it doesn't seem necessary to mix these two into another one.

Reply:
I used your suggestion here and it works without having to introduce `exterior`